right/src/usb_report_updater.c

```diff
@@ -323,7 +364,7 @@ static void applyKeyAction(key_state_t *keyState, key_action_t *action, uint8_t
             case KeyActionType_Keystroke:
                 if (keyState->previous) {
                     if (slotId == stickySlotId && keyId == stickyKeyId) {
-                        if (!IsLayerHeld() && !(secondaryRoleState == SecondaryRoleState_Triggered && IS_SECONDARY_ROLE_LAYER_SWITCHER(secondaryRole))) {
+                        if(!stickyModifierShouldStick) {
```

Comment:
May I ask why? 

I think this is better from readability standpoint, since it follows the hierarchy of the logic behind. I am afraid that merging them will make the condition resemble a magic formula.

Reply:
I think it's best to keep the control flow as simple as possible. If multiple, complex conditions are involved, I'd prefer extracting the individual conditions into easy to read variable names, then and'ing these variable names in a single if condition.

Reply:
Alright, let's try to simply merge it... I would rather not extract the condition, because it would mean inverting reading order of the entire thing, and placing the "less important" information into the higher-level scope.